common/include/pcl/impl/point_types.hpp

```diff
@@ -2402,20 +2402,87 @@ namespace traits
+                                                             pcl::fields::a,
+                                                             pcl::fields::b> >
+  { };
+
+  template <typename PointT>
+  constexpr auto has_Lab_v = has_Lab<PointT>::value;
+
+  template <typename PointT>
+  using HasLab = std::enable_if_t<has_Lab_v<PointT>, bool>;
+
+  template <typename PointT>
+  using HasNoLab = std::enable_if_t<!has_Lab_v<PointT>, bool>;
+
+  /** Metafunction to check if a given point type has HSV color space fields. */
+  template <typename PointT>
+  struct has_HSV : has_all_fields<PointT, boost::mpl::vector<pcl::fields::h,
+                                                             pcl::fields::s,
+                                                             pcl::fields::v> >
+  { };
+
+  template <typename PointT>
+  constexpr auto has_HSV_v = has_HSV<PointT>::value;
+
+  template <typename PointT>
+  using HasHSV = std::enable_if_t<has_HSV_v<PointT>, bool>;
+
+  template <typename PointT>
+  using HasNoHSV = std::enable_if_t<!has_HSV_v<PointT>, bool>;
+
   /** Metafunction to check if a given point type has either rgb or rgba field. */
   template <typename PointT>
-  struct has_color : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
-                                                              pcl::fields::rgba> >
+  struct has_rgb : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
+                                                            pcl::fields::rgba> >
+  { };
+
+  template <typename PointT>
+  constexpr auto has_rgb_v = has_rgb<PointT>::value;
+
+  template <typename PointT>
+  using HasRGB = std::enable_if_t<has_rgb_v<PointT>, bool>;
+
+  template <typename PointT>
+  using HasNoRGB = std::enable_if_t<!has_rgb_v<PointT>, bool>;
+
+  /** Metafunction to check if a given point type has any color (e.g. rgb, rgba, hsv, lab) field. */
+  template <typename PointT>
+  constexpr auto has_any_color_v = has_rgb_v<PointT> || has_HSV_v<PointT> || has_Lab_v<PointT> || has_intensity_v<PointT>;
+
+  template <typename PointT>
+  using HasAnyColor = std::enable_if_t<has_any_color_v<PointT>, bool>;
+
+  // There should be a `HasNoAnyColor` implementation which exactly means no color fields in point types,
+  // but the name seems to be obscure, so we decide to change the behavior of `HasNoColor` for this purpose
+  // as soon as the deprecation of `has_color` families in PCL 1.15.
+  // See here: https://github.com/PointCloudLibrary/pcl/pull/4761#discussion_r646427755
+
+  /** Deprecated: Metafunction to check if a given point type has either rgb or rgba field. */
+  /* 
+    Deprecations contain has_color, has_color_v, HasColor, HasNoColor. And they will be 
+    removed in PCL 1.15. Please use has_rgb, has_rgb_v, HasRGB, HasNoRGB respectively.
+  */
+  template <typename PointT>
+  struct PCL_DEPRECATED(1, 15, "Please use has_rgb instead.")
+  has_color : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
```

Comment:
using has_color = has_rgb ?

Reply:
That's quite a clean & simple practice! Thank you very much!